### PR TITLE
fix ci make target for dirs with spaces and netctl for kubeadm

### PR DESCRIPTION
**Pull request by chrisplo**

### CODE REVIEW

**Comment by chrisplo** [comment]
build PR

**Comment by chrisplo** [comment]
build PR

**Comment by chrisplo** [comment]
build PR

**Comment by chrisplo** [comment]
build PR

**Comment by chrisplo** [comment]
build PR

**Comment by unclejack** [comment]
@dseevr @neelimamukiri: Can you take a look at this, please? We should merge this with two reviews.

**Comment by unclejack** [comment]
#295 is needed to fix the CI.

File: install/ansible/install_swarm.sh [hunk 1]
```diff
@@ -162,9 +162,12 @@ fi
...
-docker run --rm --net=host $mounts $image_name sh -c "./install/ansible/install.sh $netmaster_param -a \"$ans_opts\" $install_scheduler -m $contiv_network_mode -d $fwd_mode $aci_param $cluster_param $v2plugin_param"
+mounts[0]="-v"
+mounts[1]="$(pwd)/install:/install:Z"
+mounts[2]="-v"
+mounts[3]="$(pwd)/ansible:/ansible:Z"
+mounts[4]="-v"
+mounts[5]="$src_conf_path:$container_conf_path:Z"
+mounts[6]="-v"
+mounts[7]="$(pwd)/contiv_cache:/var/contiv_cache:Z"
+docker run --rm --net=host "${mounts[@]}" $image_name sh -c "./install/ansible/install.sh $netmaster_param -a \"$ans_opts\" $install_scheduler -m $contiv_network_mode -d $fwd_mode $aci_param $cluster_param $v2plugin_param"
```
Comment (dseevr) [inline install/ansible/install_swarm.sh]: Nice 👍

File: install/k8s/install.sh [hunk 2]
```diff
@@ -276,9 +276,17 @@ echo "Applying contiv installation"
...
-chmod +x ./netctl
-rm -f /usr/bin/netctl
-cp ./netctl /usr/bin/
+
+# extract netctl from netplugin container
+echo "Extracting netctl from netplugin container"
+netplugin_version=$(
+    sed '/contiv_network_version/!d;s/.*\: \?"\(.*\)".*/\1/' \
+        install/ansible/env.json)
+docker rm netplugin-tmp >/dev/null 2>/dev/null || :
```
Comment (dseevr) [inline install/k8s/install.sh]: `&>`? `2>&1`?

File: scripts/build.sh [hunk 3]
```diff
@@ -45,7 +45,7 @@ cp -rf scripts/generate-certificate.sh $output_dir/install
 chmod +x $output_dir/install/genInventoryFile.py
```
Comment (dseevr) [inline scripts/build.sh]: Do we need to quote everywhere`$output_dir` is used?

**Comment by unclejack** [comment]
build PR

**Comment by unclejack** [comment]
build PR

File: install/k8s/install.sh (hunk 2 above)
Comment (chrisplo) [reply install/k8s/install.sh]: yeah, that's shorter,can change if the CI fails again

File: scripts/build.sh (hunk 3 above)
Comment (chrisplo) [reply scripts/build.sh]: output_dir is $(mktemp -d), which wont' have spaces . . unless TMPDIR is set and has spaces, for this PR, assuming it's not set to a dir with spaces

**Pull request merged**